### DISCUSSION TITLE
Does virtua have a better reverse mode than react-virtuoso?

### DISCUSSION THREAD

**Discussion by LookRain** [inokawa/virtua#404] [Q&A] [answered]
Hi, I'm building a chat application and looking to virtualize my message list. I was looking at react-virtuoso.

From react-virtuoso's example:
https://virtuoso.dev/prepend-items/

It looks like react-virtuoso's reverse mode is achieved with `initialTopMostItemIndex` prop, the list itself is still treated as from top to bottom.

In virtua's reverse mode, would that be the case? Can I assume that with reverse={true}, the starting position will by default stick to the bottom of the list, and list will render from the bottom to top?

Thanks so much in advance!!

**Comment by inokawa** [comment, marked as answer]
If `reverse: true` is set to virtua, the order of items are the same as react-virtuoso but items will be aligned to the end (#283).

You can prepend items using react-virtuoso with `firstItemIndex` prop. With virtua, you can do the same thing with `shift` prop. 
https://inokawa.github.io/virtua/?path=/story/advanced-chat--default
